core/model/modx/processors/security/message/getlist.class.php

```diff
@@ -1,58 +1,97 @@
-    public $permission = 'messages';
-    public $defaultSortField = 'date_sent';
-
-    public function initialize() {
-        $initialized = parent::initialize();
-        $this->setDefaultProperties(array(
-            'search' => '',
-        ));
-        return $initialized;
-    }
-    
-    public function prepareQueryBeforeCount(xPDOQuery $c) {
-        $c->innerJoin('modUser','Sender');
-        $c->where(array(
-            'recipient' => $this->modx->user->get('id'),
-        ));
-        $search = $this->getProperty('search','');
-        if (!empty($search)) {
-            $c->andCondition(array(
-                'subject:LIKE' => '%'.$search.'%',
-                'OR:message:LIKE' => '%'.$search.'%',
-            ),null,2);
-        }
-        return $c;
-    }
-
-    public function prepareQueryAfterCount(xPDOQuery $c) {
-        $c->select($this->modx->getSelectColumns('modUserMessage','modUserMessage'));
-        $c->select(array(
-            'sender_username' => 'Sender.username',
-        ));
-        return $c;
-    }
-
-    public function prepareRow(xPDOObject $object) {
-        $objectArray = $object->toArray();
-        $objectArray['sender_name'] = $object->get('sender_username');
-        $objectArray['read'] = $object->get('read') ? true : false;
-        return $objectArray;
-    }
-}
-return 'modUserMessageGetListProcessor';
\ No newline at end of file
+<?php
+
+/**
+ * Get a list of messages
+ * @param integer $start (optional) The record to start at. Defaults to 0.
+ * @param integer $limit (optional) The number of records to limit to. Defaults
+ * to 10.
+ * @param string $sort (optional) The column to sort by. Defaults to date_sent.
+ * @param string $dir (optional) The direction of the sort. Defaults to ASC.
+ * @package modx
+ * @subpackage processors.security.message
+ */
+class modUserMessageGetListProcessor extends modObjectGetListProcessor
+{
+    public $classKey = 'modUserMessage';
+    public $languageTopics = array('messages', 'user');
+    public $permission = 'messages';
+    public $defaultSortField = 'date_sent';
+
+    /**
+     * @return bool
+     */
+    public function initialize()
+    {
+        $initialized = parent::initialize();
+        $this->setDefaultProperties(array(
+            'search' => '',
+            'type' => 'inbox'
+        ));
+
+        return $initialized;
+    }
+
+    /**
+     * @param xPDOQuery $c
+     * @return xPDOQuery
+     */
+    public function prepareQueryBeforeCount(xPDOQuery $c)
+    {
+        $c->innerJoin('modUser', 'Sender');
+        $c->innerJoin('modUser', 'Recipient');
+        $c->innerJoin('modUserProfile', 'SenderProfile', 'SenderProfile.id = modUserMessage.sender');
+        $c->innerJoin('modUserProfile', 'RecipientProfile', 'RecipientProfile.id = modUserMessage.recipient');
+
+        switch ($this->getProperty('type')) {
+            case 'outbox':
+                $where = ['sender' => $this->modx->user->get('id')];
```

Comment:
@Alroniks I just noticed this short array syntax on this line, and 51 below. Any chance I could convince you to open a PR that restores PHP 5.3 compatibility?

Sorry for not spotting this sooner!